kusto/cloudinfo.go

```diff
@@ -64,8 +65,10 @@ func GetMetadata(kustoUri string, httpClient *http.Client) (CloudInfo, error) {
 		if err != nil {
 			return CloudInfo{}, err
 		}
-
-		u.Path = metadataPath
+		if !strings.HasPrefix(u.Path, "/") {
+			u.Path = "/" + u.Path
+		}
+		u = u.JoinPath(metadataPath)
```

Comment:
Do you need to prepand '/' when using url.JoinPath?

Reply:
For some reason, the "JoinPath" gives unexpected results without it.
Maybe I should add a comment?

Reply:
won't hurt